### ISSUE TITLE
nits from IoTDIR review

### ISSUE THREAD

**Issue by mcr** [anima-wg/constrained-join-proxy#53]
```
Throughout: Please pick one spelling for Join Proxy.  I see all of the
following "Join Proxy", "join Proxy", "join proxy", "Join-Proxy".
Further, sometimes is is used with "Constrained" and other times it is
used with "constrained".  I kept asking myself if there was some meaning
that I was missing in these various different spellings.  If there is
a difference, please explain it in Section 2.

Section 1: I think there is a better way to talk about "new
(unconfigured) devices."  I can think of two possible ways to better
describe the devices.  One alternative is to talk about yet-to-be-
configured devices.  The other approach would be to talk abot devices
in their factory default configuration.

Section 1: s/other IP-over-foo networks/other IP networks/

Section 1: s/is pointed at/provided/

Section 1: s/Coap/CoAP/

Section 1: s/Registrar but several hops away,/Registrar, but several hops away,/

Section 1: s/non_storing/non-storing/

Section 2: It would be better to avoid talking about the "Join
Registrar/Coordinator (JRC)" twice.  Suggest dropping the first one,
and then replacing the second with:

   The term "Registrar" is used throughout this document instead of
   "Join Registrar/Coordinator (JRC)" as defined in [[RFC8366](https://datatracker.ietf.org/doc/rfc8366/)].

Section 3: Why are there no capital letters in the section heading?

Section 3: s/IP-address/IP address/

Section 3, Figure 1: Nothing is labelled as a "Low-Power and Lossy
Network (LLN)".  The text talks about a LLN, but it is not there.

Section 3: s/Without routing/Without multi-hop routing,/

Section 5.1: s/if stateless operation /whether stateless operation/

Section 5.1.2: s/figure 10/Figure 10/

IDnits complains that there are 8 instances of too long lines in the
document, the longest one being 33 characters in excess of 72.
```

**Comment by mcr** [comment]
IDnits complains about an obsolete normative reference to [RFC 6347](https://datatracker.ietf.org/doc/rfc6347/),
which is obsoleted by [RFC 9147](https://datatracker.ietf.org/doc/rfc9147/).

**Comment by mcr** [comment]
Consistently used _Join Proxy_ in 4290ba6.

**Comment by mcr** [comment]
more fixes up to 4dfe4b3
